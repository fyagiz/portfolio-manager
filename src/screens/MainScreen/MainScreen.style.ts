import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: "#017387",
  },
  buttonContainer: {
    flexDirection: "row",
    justifyContent: "flex-end",
  },
});
