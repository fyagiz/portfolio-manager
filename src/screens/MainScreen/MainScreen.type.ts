import { ViewStyle } from "react-native";

export type MainScreenPropsType = {
  style?: ViewStyle;
};
