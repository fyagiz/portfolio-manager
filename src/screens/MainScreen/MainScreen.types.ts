import { ViewStyle } from "react-native";

export type MainScreenProps = {
  style?: ViewStyle;
};
