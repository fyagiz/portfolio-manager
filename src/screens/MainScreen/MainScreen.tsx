import { View } from "react-native";
import Dashboard from "../../components/Dashboard";
import styles from "./MainScreen.style";
import { MainScreenPropsType } from "./MainScreen.type";
import Button from "../../components/Button";
import AddAssetModal from "../../components/AddAssetModal";
import { useState } from "react";

const MainScreen = (props: MainScreenPropsType) => {
  const { style: propStyle } = props;
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <View style={[styles.container, propStyle]}>
      <AddAssetModal
        modalProps={{
          visible: modalVisible,
          transparent: true,
          animationType: "slide",
        }}
      />
      <View style={styles.buttonContainer}>
        <Button
          text="Add Asset"
          onPress={() => {
            setModalVisible(prev => !prev);
          }}
        />
      </View>
      <Dashboard />
    </View>
  );
};

export default MainScreen;
