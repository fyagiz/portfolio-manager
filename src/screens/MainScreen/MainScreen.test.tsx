import { render } from "@testing-library/react-native";
import MainScreen from "./MainScreen";

/*
1- Dashboard should be rendered.
*/

describe("MainScreen Unit Tests", () => {
  test("should render correctly", () => {
    const mainScreen = render(<MainScreen />);
    expect(mainScreen).toMatchSnapshot();
  });
});
