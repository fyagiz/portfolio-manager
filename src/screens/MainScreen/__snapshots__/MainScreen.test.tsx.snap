// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen Unit Tests should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#017387",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignSelf": "flex-start",
            "borderRadius": 4,
            "margin": 4,
            "maxWidth": 128,
            "padding": 4,
          },
          {
            "backgroundColor": "#F3CC79",
          },
        ]
      }
      testID="pressable"
    >
      <Text
        testID="buttonText"
      >
        Add Asset
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderRadius": 4,
        "margin": 2,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderRadius": 4,
            "margin": 4,
            "padding": 4,
          },
          {
            "backgroundColor": "#F3CC79",
          },
        ]
      }
      testID="pressable"
    >
      <Text
        testID="investmentName"
      >
        Test Fund
      </Text>
      <Text
        testID="profit"
      >
        5000 TL
      </Text>
      <Text
        testID="profitPercentage"
      >
        % 
        5
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderRadius": 4,
            "margin": 4,
            "padding": 4,
          },
          {
            "backgroundColor": "#F3CC79",
          },
        ]
      }
      testID="pressable"
    >
      <Text
        testID="investmentName"
      >
        Test Fund2
      </Text>
      <Text
        testID="profit"
      >
        1000 TL
      </Text>
      <Text
        testID="profitPercentage"
      >
        % 
        6
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderRadius": 4,
            "margin": 4,
            "padding": 4,
          },
          {
            "backgroundColor": "#F3CC79",
          },
        ]
      }
      testID="pressable"
    >
      <Text
        testID="investmentName"
      >
        Test Fund3
      </Text>
      <Text
        testID="profit"
      >
        1500 TL
      </Text>
      <Text
        testID="profitPercentage"
      >
        % 
        6
      </Text>
    </View>
  </View>
</View>
`;
